src/components/codemirror.tsx

```diff
@@ -43,6 +43,7 @@ export default function CodeMirror(props: CodeMirrorProps) {
 			effects: StateEffect.reconfigure.of(restoreInitialConfig())
 		});
 	};
+	setEditorTheme();
```

Comment:
This is running every render cycle, it should be inside a useEffect... which it is, but that apparently isn't working? I haven't troubleshooted myself but maybe a useSignalEffect is more appropriate? Or something else is happening that's making CodeMirror not change the theme early.

Reply:
useEffect isn't ran when the page first loads, which is causing the editor to be in light mode when reloading. It's only ran on updates. We can try useSignalEffect.

Reply:
I added one outside of the useEffect, which will run when the page first loads.

Reply:
The rest of the screen is changed to dark mode because it is stated directly in the html code, using the imported isDark varible

Reply:
@kognise

Reply:
@BrightTheBackpack you can remove the hanging `setEditorTheme` and instead add `editorRef` in the useEffect that sets the editor theme

Reply:
@BrightTheBackpack you can add a remove the hanging `setEditorTheme` and instead add `editorRef` in the useEffect where the editor theme is set.

Reply:
alternatively you can put the extensions in a list and append the dark theme if `isDark.value` is true before returning that.

Reply:
I will try your suggestions, but I'm just wondering what's wrong with the current implementation?

Reply:
I put editorRef and removed the extra setEditorTheme